sensorlib/src/main.py

```diff
@@ -0,0 +1,161 @@
+import threading
+
+import carla
+
+from src.SimulatedSensorConfigurator import SimulatedSensorConfigurator
+from src.util.SimulatedSensorUtils import SimulatedSensorUtils
+
+
+def scheduled_compute(scheduler, simulated_lidar_sensor, detection_cycle_delay_seconds):
+    scheduler.enter(detection_cycle_delay_seconds, 1, scheduled_compute)
+    simulated_lidar_sensor.compute_detected_objects()
+
+
+def main(infrastructure_id, sensor_config, noise_model_config, detection_cycle_delay_seconds, start_rpc_server,
+         xmlrpc_server_host, xmlrpc_server_port):
+    """
+    Instantiate a SimulatedLidarSensor and start an XML-RPC server to provide detected objects.
+
+    The sensor compute loop, in which detected objects are identified, is run continuously in a separate thread with
+     a detection_cycle_delay_seconds loop delay.
+
+    Two starting modes are supported:
+
+        1. With start_rpc_server=True an RPC server exposes the `get_detected_objects_json()` function to retrieve
+             the list of currently detected objects as a JSON string, and blocks further execution.
+        2. Alternatively with start_rpc_server=False, the `get_detected_objects_json()` function may be called
+            programmatically on the returned sensor object.
+
+    :param infrastructure_id: Infrastructure ID to assign to the sensor. Negative value forces auto-assignment.
+    :param sensor_config: Sensor configuration as either a file name, JSON dictionary, or blank.
+                            - If the value is a valid .yaml file name, the file is loaded.
+                            - If the value is a JSON dictionary, the dictionary is used directly.
+                            - If passed as an empty string, default values are loaded from a configuration file.
+    :param noise_model_config: Noise model configuration as either a file name, JSON dictionary, or blank.
+                            - If the value is a valid .yaml file name, the file is loaded.
+                            - If the value is a JSON dictionary, the dictionary is used directly.
+                            - If passed as an empty string, default values are loaded from a configuration file.
+    :param detection_cycle_delay_seconds: Delay in sensor computation loop (seconds).
+    :param start_rpc_server: Starts an XML-RPC server if True.
+    :param xmlrpc_server_host: Host name for XML-RPC server.
+    :param xmlrpc_server_port: Port for XML-RPC server.
+
+    :return: No return if RPC server is running as this blocks execution; The SimulatedLidarSensor object if RPC server
+                is not running.
+    """
+    # Get inputs
+    sensor_transform = carla.Transform(carla.Location(x=0.0, y=0.0, z=0.0),
+                                       carla.Rotation(pitch=0.0, yaw=0.0, roll=0.0))
+
+    # Retrieve the sensor configuration
+    if os.path.exists(sensor_config):
+        config = SimulatedSensorUtils.load_config_from_file(sensor_config)
```

Comment:
We need to make a separate PR after this to correct the current API. The sensor library should not spawn any sensors on startup and instead should be able to dynamically spawn sensors based on requests sent by the CARLA Ambassador in CDASim.

Reply:
That makes sense. One question: Are we hoping to run the sensorlib inside some other code such as the integration tool or as a standalone server?